tests/security/selinux/selinux_setup.pm

```diff
@@ -41,7 +45,10 @@ sub run {
         "setools-devel", "setools-java",      "setools-libs", "setools-tcl"
     );
     foreach my $pkg (@pkgs) {
-        zypper_call("in $pkg");
+        my $results = script_run("zypper in -y $pkg");
+        if ($results) {
+            record_soft_failure("workaround for bsc#1177655 $pkg is missing");
```

Comment:
It means if the $pkg in the list is missing, it will record as soft-fail. Will it affect the rest of them in the following tests if the $pkg is missing?

Reply:
> It means if the $pkg in the list is missing, it will record as soft-fail. Will it affect the rest of them in the following tests if the $pkg is missing?

It is hard to say as some of the pkgs are not used in the following testings.

Reply:
1. so the $pkg is the **universe set** for selinux_setup already, right?
2. my other concern is if the $pkg is missing and record soft-fail, some tests in the following could fail too. But I think it is no big deal.
Thanks,

Reply:
> 1. so the $pkg is the **universal set** for selinux_setup already, right?
yes.
> 2. my other concern is if the $pkg is missing and record soft-fail, some tests in the following could fail too. But I think it is no big deal.
I see your concerns but it is really no big deal.
>    Thanks,

Reply:
thanks ! :)